src/app/auth/account/additional-components/account-flows-tab/account-flows-tab.component.ts

```diff
@@ -0,0 +1,149 @@
+                            byDatasetIds: datasetsIds ?? [],
+                        },
+                    }),
+                    this.accountService.getAccountListFlows({
+                        accountName: this.accountName,
+                        page: 0,
+                        perPage: this.WIDGET_FLOW_RUNS_PER_PAGE,
+                        filters: { byFlowType: null, byStatus: null, byInitiator: null, byDatasetIds: [] },
+                    }),
+                    this.accountService.accountAllFlowsPaused(this.accountName),
+                ]),
+            ),
+            map(([accountListFlows, widgetListFlows, accountAllFlowsPaused]) => {
+                return { accountListFlows, widgetListFlows, accountAllFlowsPaused };
+            }),
+        );
+    }
+
+    public getPageFromUrl(): void {
+        const pageParam = this.activatedRoute.snapshot.queryParamMap.get(ProjectLinks.URL_QUERY_PARAM_PAGE);
+        if (pageParam) {
+            this.currentPage = +requireValue(pageParam);
+        }
+    }
+
+    public onPageChange(page: number): void {
+        if (page === 1) {
+            this.navigationService.navigateToOwnerView(this.accountName, AccountTabs.FLOWS);
+        } else {
+            this.navigationService.navigateToOwnerView(this.accountName, AccountTabs.FLOWS, page);
+        }
+        this.fetchTableData(page);
+    }
+
+    public onChangeFilterByStatus(status: MaybeNull<FlowStatus>): void {
+        this.fetchTableData(this.currentPage, status);
+        this.filterByStatus = status;
+    }
+
+    public onChangeFilterByInitiator(initiator: FilterByInitiatorEnum): void {
+        if (initiator !== FilterByInitiatorEnum.Account) {
+            let filterOptions: MaybeNull<InitiatorFilterInput> = null;
+            if (initiator === FilterByInitiatorEnum.System) {
+                filterOptions = { system: true };
+            }
+            this.fetchTableData(this.currentPage, this.filterByStatus, filterOptions);
+        }
+        this.filterByInitiator = initiator;
+    }
+
+    public onSearchByDatasetName(dataset: MaybeNull<Dataset>): void {
+        this.fetchTableData(this.currentPage, this.filterByStatus, null, dataset ? [dataset.id] : []);
+        this.searchByDataset = dataset;
+    }
+
+    public refreshFlow(): void {
+        this.getPageFromUrl();
+        this.fetchTableData(this.currentPage);
+    }
+
+    public onCancelFlow(params: CancelFlowArgs): void {
+        this.trackSubscription(
+            this.flowsService
+                .cancelScheduledTasks({
+                    datasetId: params.datasetId,
+                    flowId: params.flowId,
+                })
+                .subscribe((success: boolean) => {
+                    if (success) {
+                        setTimeout(() => {
+                            this.refreshFlow();
+                        }, this.TIMEOUT_REFRESH_FLOW);
+                    }
+                }),
+        );
+    }
```

Comment:
This method is identical in both components, move to base class.
Generally, think how you can generalize most of the algorithms so that the code is mostly written once with some abstraction in the base class and some overrides in derived classes.